### ISSUE TITLE
trying to re-adapt your code for the study of long 1-D sequences

### ISSUE THREAD

**Issue by chrisclarkson** [eriklindernoren/Keras-GAN#79]
Hi,
Your code is very interesting to me. 
I have sequences of long, sparse 1_D vectors (10000 digits, made of of 0s and 1s) that I am trying to classify. I have previously implemented a simple CNN to classify them with relative success.

```
def create_shallow_model(shape,repeat_length,stride):
    model = Sequential()
    model.add(Conv1D(75,repeat_length,strides=stride,padding='same', input_shape=shape, activation='relu'))
    model.add(MaxPooling1D(repeat_length))
    model.add(Flatten())
    model.add(Dense(1, activation='sigmoid'))
    model.compile(loss='binary_crossentropy', optimizer='rmsprop', metrics=['accuracy'])
    return model
```

I am trying to create a GAN that returns a sequence that is emblematic of a sequence that have a '1' label (or in other words the sequence that is distinguishing the '1' labels from the '0's)...

I have pieced together what I can from the code in your 'DCGAN' repository and another GitHub page: https://github.com/osh/KerasGAN/blob/master/MNIST_CNN_GAN_v2.ipynb.

I have attempted to re-implement your analysis of 2D tensors on my 1-D data.

However I am struggling with the generator's design and what type of data I'm supposed to give it. Thus far I have managed to get a functioning version of your code. See below the best version of a generator that I could manage:

```
def build_generator(shape):
	model = Sequential()
	model.add(Conv1D(128, kernel_size=3, padding="same",input_shape=shape)) #start with conv
	model.add(BatchNormalization(momentum=0.8))
	model.add(Activation("relu"))
	model.add(UpSampling1D())
	model.add(Conv1D(64, kernel_size=3, padding="same"))
	model.add(BatchNormalization(momentum=0.8))
	model.add(Activation("relu"))
	model.add(UpSampling1D())
	model.add(Conv1D(64, kernel_size=3, padding="same"))
	model.add(BatchNormalization(momentum=0.8))
	model.add(Activation("tanh"))
	model.add(Conv1D(1, kernel_size=3, padding="same"))
	#model.add(Activation("tanh"))
	model.summary()
	return model
```
I intentionally skip the first 2 layers that you use:
```
model.add(Dense(train_X.shape[1], activation="relu", input_dim=self.latent_dim))
model.add(Reshape((1, train_X.shape[1], 128)))
```
(This is because I have tried in the past to pass 1 D data to a network that starts with fully connected layers and passes to convolutional layers but this is difficult as reshaping the input to the convolutions in a way that they will be accepted is difficult).

Hence my generator is just an upsampling network that starts with a vector 1/4 the size of the final output (which will ultimately be 10000 elements long for the discriminator to accept):

```
noise = np.random.normal(0, 1, (batch_size, int(train_X.shape[1]/4)))
noise=noise.reshape(noise.shape[0], noise.shape[1],1)
generator = build_generator(noise.shape[1:])

```
and then I build the discriminator:

```
discriminator=load_model('....load_old_model....')
discriminator.trainable = False

```
And so then I add the layers from you the generator to discriminator:

```
def add_layers(net1,net2):
	new=Sequential()
	counter=1
	for l in net1.layers:
		print(counter)
		new.add(l)
		counter+=1
	for l in net2.layers:
		print(counter)
		new.add(l)
		counter+=1
	return new

combined=add_layers(generator,discriminator)
combined.compile(loss='binary_crossentropy', optimizer=optimizer)
```

Then I run your code to train the GAN (note that I only give it data that had a label '1' in the original analysis- `train_X[train_y==1]` as I want a sequence that is representative of the sequences that had a '1' label):

```
epochs=70
batch_size=32
save_interval=50
valid = np.ones((batch_size, 1))
fake = np.zeros((batch_size, 1))
for epoch in range(epochs):
	# ---------------------
	#  Train Discriminator
	# ---------------------
	# Select a random half of images
	idx = np.random.randint(0, train_X[train_y==1].shape[0], batch_size)
	imgs = train_X[train_y==1][idx]
	# Sample noise and generate a batch of new images
	noise = np.random.normal(0, 1, (batch_size, int(train_X.shape[1]/4)))
	noise=noise.reshape(noise.shape[0], noise.shape[1],1)
	gen_imgs = generator.predict(noise)
	# Train the discriminator (real classified as ones and generated as zeros)
	d_loss_real = discriminator.train_on_batch(imgs, valid)
	d_loss_fake = discriminator.train_on_batch(gen_imgs, fake)
	d_loss = 0.5 * np.add(d_loss_real, d_loss_fake)
	# ---------------------
	#  Train Generator
	# ---------------------
	# Train the generator (wants discriminator to mistake images as real)
	g_loss = combined.train_on_batch(noise, valid)
	# Plot the progress
	print ("%d [D loss: %f, acc.: %.2f%%] [G loss: %f]" % (epoch, d_loss[0], 100*d_loss[1], g_loss))
	# If at save interval => save generated image samples
	if epoch % save_interval == 0:
		save_imgs(epoch)
```
Although the code runs, I am suspicious that the results are viable.....

The generator's learning loss is 0 from the very start:
```
0 [D loss: 0.000041, acc.: 100.00%] [G loss: 0.000000]
1 [D loss: 0.000026, acc.: 100.00%] [G loss: 0.000000]
2 [D loss: 0.000025, acc.: 100.00%] [G loss: 0.000000]
3 [D loss: 0.000018, acc.: 100.00%] [G loss: 0.000000]
4 [D loss: 0.000016, acc.: 100.00%] [G loss: 0.000000]
5 [D loss: 0.000017, acc.: 100.00%] [G loss: 0.000000]
6 [D loss: 0.000013, acc.: 100.00%] [G loss: 0.000000]
7 [D loss: 0.000013, acc.: 100.00%] [G loss: 0.000000]
8 [D loss: 0.000011, acc.: 100.00%] [G loss: 0.000000]
9 [D loss: 0.000050, acc.: 100.00%] [G loss: 0.000000]
....
```
I also tried creating pseudo data that was more like the type that I would usually feed the discriminator but this made little difference to the generator's loss....

```
def make_fake(rows):
	n=[]
	for i in range(rows):
		noise1=np.zeros(int(train_X.shape[1]/8))
		noise2=np.ones(int(train_X.shape[1]/8))
		noise=np.hstack((noise1,noise2))
		np.random.shuffle(noise)
		n.append(noise)
	n=np.array(n)
	n=n.reshape(n.shape[0], n.shape[1],1)
	return n

noise=make_fake(batch_size)

0 [D loss: 0.093675, acc.: 100.00%] [G loss: 0.000811]
1 [D loss: 0.055896, acc.: 100.00%] [G loss: 0.313676]
2 [D loss: 0.037136, acc.: 100.00%] [G loss: 0.000000]
3 [D loss: 0.030188, acc.: 100.00%] [G loss: 0.000000]
4 [D loss: 0.021618, acc.: 100.00%] [G loss: 0.000000]
5 [D loss: 0.018344, acc.: 100.00%] [G loss: 0.000000]
6 [D loss: 0.016752, acc.: 100.00%] [G loss: 0.000000]
7 [D loss: 0.011288, acc.: 100.00%] [G loss: 0.000000]
8 [D loss: 0.010168, acc.: 100.00%] [G loss: 0.000000]
```

I would have hoped for the generators loss to take a lot longer to reach 0...

Any way after 50 epochs the output of the generator looked like this:

![epoch_50](https://user-images.githubusercontent.com/26774207/44843930-71638780-ac41-11e8-9c49-16811b434ee6.png)

when I more hoped it would look something like this exemplary sequence:

![14_centered1000shallow_top_scoring](https://user-images.githubusercontent.com/26774207/44843969-893b0b80-ac41-11e8-9070-23c4b30604af.png)

Can you help me set up a GAN that creates sequences that are representative of what my original, very simple CNN was recognising as samples with 1 labels?
